hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java

```diff
@@ -329,11 +438,22 @@ protected void cleanup(Context context) {
           LOG.error("fail to close replicated connection in cleanup", e);
         }
       }
+      if (reCompareExecutor != null && !reCompareExecutor.isShutdown()) {
+        reCompareExecutor.shutdown();
+        try {
+          boolean terminated = reCompareExecutor.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
This `awaiTermination` sequence is net new. The test was missing some invalid rows, it was due to the us not waiting on the executor shutdown.